### DISCUSSION TITLE
Раздел Нерабочее время дополнить производственным календарём

### DISCUSSION THREAD

**Discussion by 112358aalexx** [mikopbx/Core#53] [Ideas (Идеи)]
Добрый день!
Предлагаю реализовать свой производственный календарь (или использовать любой из имеющихся сторонних по его API) для определения рабочий ли день и более гибкого формирования правил маршрутизации звонков на IVR компании. 
Этим вы избавите от ненужной возьни с кастомизацией под государственные праздники с переносами рабочих дней.

**Comment by zhenterik** [comment]
Хорошая идея, поддерживаю!

**Reply by matveevandrey** [reply]
+1 

Раньше на "голом астериске" реализовывали так:

; check is day off online
same => n,Set(ISDAYOFF=${CURL(http://isdayoff.ru/today?cc=ru)})
same => n,Verbose("${ISDAYOFF}")
same => n,GotoIf($["${ISDAYOFF}" = "1"]?closed:open)

same => n(open),GotoIfTime(8:59-18:55,*,*,*?incoming-open,хххххх,1)
same => n(closed),Goto(incoming-closed,хххххх,1)

**Reply by boffart** [reply]
такое и на mikopbx можно сделать через "кастомизацию системных файлов"
добавить в конце файла:
```
[add-trim-prefix-clid-custom]
exten => _[0-9*#+a-zA-Z][0-9*#+a-zA-Z]!,1,NoOp()
  same => n,Set(ISDAYOFF=${CURL(http://isdayoff.ru/today?cc=ru)})
  same => n,Verbose("${ISDAYOFF}")
  same => n,GotoIf($["${ISDAYOFF}" = "1"]?closed:open)
  
  same => n(open),GotoIfTime(8:59-18:55,,,*?incoming-open,хххххх,1)
  same => n(closed),Goto(incoming-closed,хххххх,1)
  same => n,return
```

**Comment by matveevandrey** [comment]
+1
